lib/credentials/shared_ini_file_credentials.js

```diff
@@ -87,6 +95,70 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
   /**
    * @api private
    */
+  loadRoleProfile: function loadRoleProfile(creds, callback) {
+    var self = this;
+    var roleArn = creds[this.profile]['role_arn'];
```

Comment:
It might be a good idea to store `creds[this.profile]` in a variable since we're calling it so frequently within the same function.